debugAdapterProtocol.json

```diff
@@ -2574,6 +2582,10 @@
 							"memoryReference": {
 								"type": "string",
 								"description": "A memory reference to a location appropriate for this result.\nFor pointer type eval results, this is generally a reference to the memory address contained in the pointer.\nThis attribute may be returned by a debug adapter if corresponding capability `supportsMemoryReferences` is true."
+							},
+							"locationReference": {
```

Comment:
🤔 which DAP call actually backs the "Watch" pane? Is this backed by the `EvaluateResponse`?

If so, I think we do want to have both `declarationLocation` and `valueLocation` here. 

In the UI this would probably be useful for watched expressions: For simple expressions such as `my_variable.member1`, the user could still use the "go to declaration" functionality also from the "Watch" pane. (Of course, for more complex watch expressions like `my_variable.some_method() == 1`, there would be no `declarationLocation` available)

Reply:
Yea in VS Code the watch pane runs `evaluate` with `context = 'watch'`

Reply:
> I think it's less likely DA's would support declarationLocation in these cases since that requires language server intelligence to understand expressions

While looking into how to implement this for lldb-dap, I realized that LLDB will likely be able to provide a `declarationLocation` also for `EvaluateRequest`. See [this comment](https://github.com/llvm/llvm-project/blob/019ef522756886caa258daf68d877f84abc1b878/lldb/tools/lldb-dap/lldb-dap.cpp#L1377).

The current PR does not yet introduce it for `EvaluateResponse`, though. Do you think it makes sense to also add it for `EvaluateExpression`? (both here and in VS-Code)